### ISSUE TITLE
Catch Image.asset not found error

### ISSUE THREAD

**Issue by surjithctly** [bundlesandbatches/next-sanity-image#14]
I'm getting the following error when we pass `image.alt` but not `image.asset`. This plugin should send appropriate error when `image.asset` is not found. 

![image](https://user-images.githubusercontent.com/1884712/126159192-139269a9-719d-4352-8d71-90cad6bd3dba.png)

In my code, I cannot do the following because hooks cannot call conditionally. 

```js
// This won't work
// Not able to call hooks conditionally. 

  if (!image.asset) {
    return null;
  }
 const imageProps = useNextSanityImage(client, image, {
    imageBuilder: CustomImageBuilder
  });
```

**Comment by lorenzodejong** [comment]
Could you perhaps clarify in which scenario this occurs, preferably with a code example of how you retrieve the image from Sanity?

**Comment by lorenzodejong** [comment]
Closing due to inactivity, issue can be reopened if we can get a reproducible case.

**Comment by surjithctly** [comment]
Hey, 

Sorry, missed your comment. It happens when "`image`" exists but not "`image.asset`". I had this issue when I added an image caption without adding image. 

To Reproduce:
1. Go to Sanity and create and `image` field with caption / alt
2. Now fill the alt & caption, but don't upload any photo
3. Save & run the plugin 
4. It will throw this error because the `image` does not have `asset`.

**Comment by surjithctly** [comment]
Now, I tried to add an exception in my code for this scenario. 

```
 if (!image.asset) {
    return null;
  }
```

But now there is another error called: `hooks cannot call conditionally.`

So, I can't do this error exception in my code. So checking the possibility to check this in the plugin itself.

**Comment by lorenzodejong** [comment]
Just checked this in the Sanity studio which i use for testing purposes. It looks like i have to select/upload an image before the "Edit details" button becomes visible and i can specify a caption/alt. What does the configuration of your Sanity image field look like?

**Comment by surjithctly** [comment]
I have `isHighlighted` on as per https://www.sanity.io/docs/image-type

So it will be visible all time.

**Comment by lorenzodejong** [comment]
Thanks for the extensive example. I've just released version 3.1.7 which includes a check for the existence of the image id in order to fix this issue.
